src/util.cpp

```diff
@@ -6139,39 +6139,82 @@ std::vector<std::string> splitString(const std::string& str)
  * such that it does not remove non-spacers
  */ 
 QCString stripLeft(const QCString& text)
-{
+{  
   auto lines = splitString(text.str());
+  
+  // to be used in case of implement SNIPPET_TRIMLEFT = TABCONVERT
+  // auto tabSize = Config_getInt(TAB_SIZE); 
```

Comment:
Why not always do the `detabbing`?

In `markdown.cpp` there is a method `QCString Markdown::detab(const QCString &s,int &refIndent)` so it might be useful to reuse this routine (as it calculates the minimum start spacing and translates the `tabs` to spaces).
So do something like first detab the original text and on the result perform the trimming.

Reply:
thanks you for your double checks:
- enum is used for future extension and specialization of behavior as you noted
- replaced `(for `<tabs>` also the setting of  \ref cfg_tab_size "TAB_SIZE" is taken into consideration.` because effectively the "YES" option would not consider TAB_SIZE actually it will be used eventually for a mixed mode TABCONVERT
- I take a look to the `QCString Markdown::detab(const QCString &s,int &refIndent)`

Reply:
Tried `detab` but doesn't work for the need, it doesn't remove spaces ( [see here](https://github.com/doxygen/doxygen/blob/aed5276eb131252a6d4eaf4b00ee145127541f1f/src/markdown.cpp#L3227) )

![image](https://user-images.githubusercontent.com/13405008/215268106-3c9331f7-80fb-4fea-bf8a-917fed66a7ac.png)

Reply:
You return the `detab` text but this is just changing tabs into spaces and determines the `refIndent`, this should be fed into `stripLeft` and here the "superfluous" spaces at the left could be removed.

Reply:
You right, but effectively seems to me a forcing of modularization, actually proposed PR do the thing producing expected result, of course if you feel that this is necessary you could change as your need. (note: detab is a private method and the environment which planned to be used is for markdown text, future changes to its behavior could brake desired effect of SNIPPER_TRIMLEFT behavior ).
One thing that missed is the unit test, when I have some space I will add if you consider the PR could ok.